fedimint-testing/src/btc/real.rs

```diff
@@ -175,14 +179,26 @@ impl RealBitcoinTest {
 /// locked version - locks the global lock during construction
 pub struct RealBitcoinTestLocked {
     inner: RealBitcoinTestNoLock,
-    _guard: tokio::sync::MutexGuard<'static, ()>,
+    _guard: fs_lock::FileLock,
 }
 
 #[async_trait]
 impl BitcoinTest for RealBitcoinTest {
     async fn lock_exclusive(&self) -> Box<dyn BitcoinTest + Send + Sync> {
         trace!("Trying to acquire global bitcoin lock");
-        let _guard = REAL_BITCOIN_LOCK.lock().await;
+        let _guard = tokio::task::block_in_place(|| {
+            let lock_file_path = std::env::temp_dir().join("fm-test-bitcoind-lock");
+            fs_lock::FileLock::new_exclusive(
+                std::fs::OpenOptions::new()
+                    .write(true)
+                    .create(true)
+                    .truncate(true)
+                    .open(&lock_file_path)
```

Comment:
Use std::fs::create, it does the same under the hood

Reply:
We have a semgrep lint for it, to force people to think about truncation behavior, as it was used incorrectly in the past. I don't mind longer version.